### PR TITLE
plasma5: set default session to plasma X11 and fix sddm

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change

As suggested by @ttuegel in https://github.com/NixOS/nixpkgs/pull/139657#issuecomment-951393788 and many others in #143272, set the default session to plasma X11 instead of wayland for now.

`sddm` upstream currently doesn't support system-wide default session (https://github.com/sddm/sddm/issues/326), I implemented it in a patch in this PR. I'll maybe upstream it later if time permits.
Note that if you logined non-default session once, it will be selected in the next time. That is, the last logined session still takes precedence over system-wide default session.

Tested in nixos vm.

cc: @abbradar @tobiasBora 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/display-managers/sddm.nix [hunk 1]
```diff
@@ -30,6 +30,10 @@ let
       HaltCommand = "/run/current-system/systemd/bin/systemctl poweroff";
       RebootCommand = "/run/current-system/systemd/bin/systemctl reboot";
       Numlock = if cfg.autoNumlock then "on" else "none"; # on, off none
+
+      # Set default session.
```
Comment (SuperSandro2000) [inline nixos/modules/services/x11/display-managers/sddm.nix:34]:
```suggestion
```

File: nixos/modules/services/x11/display-managers/sddm.nix [hunk 2]
```diff
@@ -30,6 +30,10 @@ let
...
+      # See pkgs/applications/display-managers/sddm/sddm-default-session.patch
```
Comment (SuperSandro2000) [inline nixos/modules/services/x11/display-managers/sddm.nix:35]:
```suggestion
      # Implementation is done via pkgs/applications/display-managers/sddm/sddm-default-session.patch
```

File: nixos/modules/services/x11/display-managers/default.nix [hunk 3]
```diff
@@ -280,7 +280,8 @@ in
             null;
         example = "gnome";
         description = ''
-          Graphical session to pre-select in the session chooser (only effective for GDM and LightDM).
+          Graphical session to pre-select in the session chooser (only effective for GDM, LightDM
+          and SDDM).
```
Comment (SuperSandro2000) [inline nixos/modules/services/x11/display-managers/default.nix:284]:
```suggestion
          Graphical session to pre-select in the session chooser (only effective for GDM, LightDM and SDDM).
```

**Review by SuperSandro2000 — COMMENTED** [review]
someone who knows cpp please look at the patch

**Review by ttuegel — APPROVED** [review]

**Comment by ttuegel** [comment]
Thanks!

**Pull request merged**